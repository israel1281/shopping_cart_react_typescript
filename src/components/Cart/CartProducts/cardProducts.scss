.cartProductsContainer {
  position: relative;
  min-height: 280px;
  padding-bottom: 200px;

  .cartProductsEmpty {
    color: #ececec;
    text-align: center;
    line-height: 40px;
    font-weight: 500;
  }
}
