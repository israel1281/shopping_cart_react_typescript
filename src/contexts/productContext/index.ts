export { ProductsProvider } from "./ProductContextProvider";
export { default as useProducts } from "./useProducts";
