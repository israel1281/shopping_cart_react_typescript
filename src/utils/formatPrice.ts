const formatPrice = (price: number, currencyId: string): string => {
  switch (currencyId) {
    case "NGN":
      return price.toFixed(4).replace(".", ",");
    default:
      return price.toFixed(2);
  }
};

export default formatPrice;
