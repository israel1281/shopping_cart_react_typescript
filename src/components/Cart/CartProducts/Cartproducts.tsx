import { ICartProduct } from "../../../models";
import CartProduct from "./CartProduct";
import "./cardProducts.scss";

interface IProps {
  products: ICartProduct[];
}

const CartProducts = ({ products }: IProps) => {
  return (
    <div className="cartProductsContainer">
      {products?.length ? (
        products.map((p) => <CartProduct product={p} key={p.sku} />)
      ) : (
        <p className="cartProductsEmpty">
          Add some products in the cart <br />
          ðŸ™‚
        </p>
      )}
    </div>
  );
};

export default CartProducts;
