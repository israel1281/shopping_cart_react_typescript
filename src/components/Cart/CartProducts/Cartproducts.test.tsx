import { CartProvider } from "../../../contexts/cartContext";
import { mockCartProducts } from "../../../utils/mockData";

import CartProducts from ".";

describe("[CartProducts Component", () => {
  const setup = (props = {}) => {
    return (
      <CartProvider>
        <CartProducts products={mockCartProducts} {...props} />
      </CartProvider>
    );
  };

  test("should render correctly", () => {
    const view = setup();
    expect(view).toMatchSnapshot();
  });
});
