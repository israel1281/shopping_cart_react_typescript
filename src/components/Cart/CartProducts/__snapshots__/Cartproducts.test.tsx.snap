// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CartProducts Component should render correctly 1`] = `
<CartProvider>
  <CartProducts
    products={
      Array [
        Object {
          "availableSizes": Array [
            "S",
            "L",
            "XL",
            "XXL",
          ],
          "currencyFormat": "$",
          "currencyId": "USD",
          "description": "14/15 s/nº",
          "id": 0,
          "installments": 9,
          "isFreeShipping": true,
          "price": 10.9,
          "quantity": 1,
          "sku": 8552515751438644,
          "style": "White T-shirt",
          "title": "Cropped Stay Groovy off white",
        },
        Object {
          "availableSizes": Array [
            "X",
            "ML",
            "L",
          ],
          "currencyFormat": "$",
          "currencyId": "USD",
          "description": "",
          "id": 11,
          "installments": 3,
          "isFreeShipping": true,
          "price": 13.25,
          "quantity": 1,
          "sku": 39876704341265610,
          "style": "Wine",
          "title": "Basic Cactus White T-shirt",
        },
        Object {
          "availableSizes": Array [
            "XL",
          ],
          "currencyFormat": "$",
          "currencyId": "USD",
          "description": "14/15 s/nº - Jogador",
          "id": 4,
          "installments": 12,
          "isFreeShipping": false,
          "price": 25.9,
          "quantity": 1,
          "sku": 9197907543445676,
          "style": "Tony Hawk",
          "title": "Skater Black Sweatshirt",
        },
      ]
    }
  />
</CartProvider>
`;
